charmhelpers/core/host.py

```diff
@@ -114,6 +114,33 @@ def service_stop(service_name, **kwargs):
     return service('stop', service_name, **kwargs)
 
 
+def service_enable(service_name, **kwargs):
+    """Enable a system service.
+
+    The specified service name is managed via the system level init system.
+    Some init systems (e.g. upstart) require that additional arguments be
+    provided in order to directly control service instances whereas other init
+    systems allow for addressing instances of a service directly by name (e.g.
+    systemd).
+
+    The kwargs allow for the additional parameters to be passed to underlying
+    init systems for those systems which require/allow for them. For example,
+    the ceph-osd upstart script requires the id parameter to be passed along
+    in order to identify which running daemon should be restarted. The follow-
+    ing example restarts the ceph-osd service for instance id=4:
+
+    service_enable('ceph-osd', id=4)
+
+    :param service_name: the name of the service to enable
+    :param **kwargs: additional parameters to pass to the init system when
+                     managing services. These will be passed as key=value
+                     parameters to the init system's commandline. kwargs
+                     are ignored for init systems not allowing additional
+                     parameters via the commandline (systemd).
```

Comment:
This is misleading; the `**kwargs` aren't actually used in the function, but this docstring implies that they are.  If `**kwargs` are added as a compatibility, then this docstring should say that they are ignored.  In which case, why accept them at all?  I suspect that they shouldn't be included in the function signature at all?

Reply:
Hello Alex. Thanks for the review. The `**kwargs` in the function is indeed solely for compatibility. Do you suggest to remove it completely or just updating the signature?

Reply:
I think, as it's a new function, the `**kwargs` should just be dropped from the signature and the docstring; they don't do anything and it would be an error if a user of the function called it with a kwarg expecting it to do something with it.  Much better to fail hard, rather than silently pass but perhaps propagate a semantic error further into the run time of the code.

Reply:
Sounds great. I have pushed a new commit addressing out discussion. I also noticed similar issues with service_restart function. I will open another PR afterwards.